src/reducetensor.cpp

```diff
@@ -413,7 +544,9 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
         MIOPEN_THROW("The indices size allocated is not enough!");
 
     // void* ws_buf1_global = static_cast<void*>(workspace);
-    Data_t ws_buf1_global     = workspace;
+    Data_t ws_buf1_global = miopen::IsEnabled(MIOPEN_DISABLE_DYNAMIC_REDUCTION{})
+                                ? workspace
+                                : static_cast<char*>(workspace) + 4096;
```

Comment:
error: invalid static_cast from type ‘Data_t {aka _cl_mem*}’ to type ‘char*’
Without checking too much details into the code, would reinterpret_cast be more appropriate?

Reply:
This has been fixed.   With OpenCL backend, the Data_t is clmem_t *,  reinterpret_cast<> should work.  But to be compatible with both HIP and OpenCL,  I should other method.